ravedude/src/main.rs

```diff
@@ -91,12 +96,53 @@ fn main() {
 }
 
 fn ravedude() -> anyhow::Result<()> {
-    let args: Args = structopt::StructOpt::from_args();
+    let mut args: Args = structopt::StructOpt::from_args();
+    if args.dump_config {
+        println!(
+            "{}",
+            toml::to_string(&board::get_board(args.board.as_deref())?)?
+        );
+        return Ok(());
+    }
+
+    let ravedude_config_exists = std::path::Path::new("Ravedude.toml").try_exists()?;
+
+    if ravedude_config_exists {
+        if let Some(board) = args.board.take() {
+            if args.bin.is_none() {
+                // The board arg is taken before the binary, so rearrange the args when Ravedude.toml exists
+                args.bin = Some(std::path::PathBuf::from(board));
+            } else {
+                anyhow::bail!("can't pass board as command-line argument when Ravedude.toml is present; set `board = \"{}\"` under [general] in Ravedude.toml", board)
+            }
+        }
+    } else {
+        warning!(
+            "Passing the board as command-line argument is deprecated, use Ravedude.toml instead:\n\n# Ravedude.toml\n{}",
+            toml::to_string(&config::RavedudeConfig::from_args(&args)?)?
+        );
```

Comment:
This warning is also shown when no commandline arguments are passed at all (and no `Ravedude.toml` is present.  This is a bit confusing, instead we should show an error about `Ravedude.toml` being missing.

Reply:
Fixed :+1: